component/src/main/java/org/wso2/extension/siddhi/io/jms/sink/JMSPublisher.java

```diff
@@ -65,17 +65,20 @@ public void run() {
     }
 
     private CarbonMessage handleCarbonMessage(Object payload) throws UnsupportedEncodingException {
-
+        StringBuilder sb = new StringBuilder();
         if (payload instanceof String) {
             CarbonMessage textCarbonMessage = new TextCarbonMessage(payload.toString());
             this.jmsProperties.put(MESSAGE_TYPE_FIELD, TEXT_MESSAGE_TYPE);
             return textCarbonMessage;
         } else if (payload instanceof Map) {
-            MapCarbonMessage mapCarbonMessage = new MapCarbonMessage();
-            ((Map) payload).forEach((key, value) -> {
-                mapCarbonMessage.setValue((String) key, (String) value);
-            });
-            return mapCarbonMessage;
+                MapCarbonMessage mapCarbonMessage = new MapCarbonMessage();
+                ((Map) payload).forEach((key, value) -> {
+                    sb.setLength(0);
+                    sb.append(value);
+                    String valueOfKey = sb.toString();
```

Comment:
why are you doing this?

Reply:
To cast any type of variable to string

Reply:
Is this to avoid ClassCastExceptions. If so let's use String.valueOf